lib/upload.js

```diff
@@ -3,15 +3,29 @@ var AWS = require('aws-sdk');
 AWS.config.region = process.env.AWS_REGION; // set your Environment Variables...
 var lambda = new AWS.Lambda();
 var fs = require('fs');                     // so we can read the zip file
+var assert = require('assert');
 var utils = require('../lib/utils');
 var base_path = utils.get_base_path();
-var pkg = require(base_path + 'package.json');
-var version = pkg.version;
-var functionName = pkg.name + '-v' + version.substring(0, version.indexOf('.'));
-var zip_file_path = process.env.TMPDIR + pkg.name + '.zip';
-var assert = require('assert');
 
-module.exports = function upload (callback) {
+/**
+ * upload does exacatly what you expect: uploads your lambda as a .zip to AWS!
+ * @param {Object} pkg - the package.json for the function you want to deploy
+ * this allows us to have a multi-lambda GitHub Repo or to deploy several Lambdas
+ * in one command by specifying the a different package.json for each of them.
+ * @param {Function} callback - called once the Lambda has been uploaded
+ */
+module.exports = function upload (pkg, callback) {
```

Comment:
@nikhilaravi good question for now this is only used *internally* it will be exposed as a user feature in V4. 👍